### PR TITLE
Rename unexpected containers whose name conflicts with mistry generated names

### LINKED ISSUE #20: Implement a strategy for stale containers
Right now, we remove containers in `StartContainer()` after they run successfully. However, it might happen, for example in case of a mistry failure while a container was running (eg. `SIGKILL`), that containers are left around with nobody to remove them.

We should decide on how to treat such cases.

### PR DESCRIPTION
**Pull request by dtheodor**
Closes #20

### CODE REVIEW

File: cmd/mistryd/job.go [hunk 1]
```diff
@@ -233,6 +252,20 @@ func (j *Job) StartContainer(ctx context.Context, cfg *Config, c *docker.Client,
 	return result.State.ExitCode, nil
 }
 
+var conflictRe = regexp.MustCompile(`Conflict. The container name "\S+" is already in use by container "(\S+)".`)
```
Comment (agis) [inline cmd/mistryd/job.go]: I'm concerned that this is too fragile and non-future-proof. Can't we just check if the container exists and if it does, delete it?

Comment (agis) [reply cmd/mistryd/job.go]: And they don't expose the helper or something we could reuse?

Comment (dtheodor) [reply cmd/mistryd/job.go]: I switched to a `renameIfExists` before creation. I'm renaming, I'm not sure we want to disrupt any existing containers. Any rogue containers will be garbage collected with our periodic prune

File: cmd/mistryd/job.go [hunk 2]
```diff
@@ -233,6 +237,35 @@ func (j *Job) StartContainer(ctx context.Context, cfg *Config, c *docker.Client,
...
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func randomHexString() string {
+	buf := make([]byte, 16)
+	rand.Read(buf)
```
Comment (agis) [inline cmd/mistryd/job.go]: Error check?

Comment (agis) [reply cmd/mistryd/job.go]: Keep in mind that the default source you're using here...
> [...] produces a deterministic sequence of values each time a program is run.

(source: https://golang.org/pkg/math/rand)

Comment (dtheodor) [reply cmd/mistryd/job.go]: rand.Read can never fail, not sure why they decided to return an error, maybe to keep the same API with other Read functions?

Comment (dtheodor) [reply cmd/mistryd/job.go]: Ok, so if we restart the service, the same sequence of "random" values will be produced. I think this is OK. I could also use crypto/rand Read()

Comment (dtheodor) [reply cmd/mistryd/job.go]: There's also this recipe `rand.Seed( time.Now().UTC().UnixNano())`

Comment (agis) [reply cmd/mistryd/job.go]: This seems like a decent choice. You only need to do this once when the server boots though.

Comment (dtheodor) [reply cmd/mistryd/job.go]: I'm initializing on mistry main() now, so any usage of rand called from the proper entrypoint (i.e. not from the tests) will be properly seeded

**Pull request merged**